modules/extensions/src/api/experimental/exec.ts

```diff
@@ -52,18 +53,26 @@ export async function exec(
     })
   );
 
-  await promise;
+  const result: Promise<SeparatedOutputExecResult | CombinedOutputExecResult> =
+    new Promise((resolve) => {
+      promise.then(() => {
```

Comment:
we want to use the resolved value of this promise to get the exitError (which i also renamed to exitCode in #72)

Reply:
Problem with waiting for the promise to be resolved is that you can't dispose the process.  If you somehow get into an infinitely-looping process like `tsc --watch`, the promise will never resolve.  Won't the exit code get returned with the promise?

Reply:
a running process has no exit code, so it makes sense that we never resolve the `result`

also we aren't supplying the exitCode anywhere yet, it's just set to an empty string